evennia/contrib/grid/basicmap/basicmap.py

```diff
@@ -0,0 +1,242 @@
+shown on the map).
+
+An example map:
+```
+       |
+     -[-]-
+       |
+       |
+-[-]--[-]--[-]--[-]
+  |    |    |    |
+       |    |    |
+     -[-]--[-]  [-]
+       | \/ |    |
+     \ | /\ |
+     -[-]--[-]
+```
+
+Installation:
+
+Adding the `BasicMapCmdSet` to the default character cmdset will add the `map` command.
+
+Specifically, in `mygame/commands/default_cmdsets.py`:
+
+```
+...
+from evennia.contrib.grid.basicmap import basicmap  # <---
+
+class CharacterCmdset(default_cmds.Character_CmdSet):
+    ...
+    def at_cmdset_creation(self):
+        ...
+        self.add(basicmap.BasicMapCmdSet)  # <---
+
+```
+
+Then `reload` to make the new commands available.
+
+Additional Settings:
+
+In order to change your default map size, you can add to `mygame/server/settings.py`:
+
+BASIC_MAP_SIZE = 5
+
+This changes the default map width/height. 2-5 for most clients is sensible.
+
+If you don't want the player to be able to specify the size of the map, ignore any
+arguments passed into the Map command.
+"""
+import time
+from django.conf import settings
+from evennia import CmdSet
+from evennia.commands.default.muxcommand import MuxCommand
+
+_BASIC_MAP_SIZE = settings.BASIC_MAP_SIZE if hasattr(settings, 'BASIC_MAP_SIZE') else 2
+
+# _COMPASS_DIRECTIONS specifies which way to move the pointer on the x/y axes and what characters to use to depict the exits on the map.
+_COMPASS_DIRECTIONS = {
+    'north': (0, -3, ' | '),
+    'south': (0, 3, ' | '),
+    'east': (3, 0, '-'),
+    'west': (-3, 0, '-'),
+    'northeast': (3, -3, '/'),
+    'northwest': (-3, -3, '\\'),
+    'southeast': (3, 3, '\\'),
+    'southwest': (-3, 3, '/'),
+    'up': (0, 0, '^'),
+    'down': (0, 0, 'v')
+}
+
+
+class Map(object):
+    def __init__(self, caller, size=_BASIC_MAP_SIZE, location=None):
+        self.start_time = time.time()
+        self.caller = caller
+        self.max_width = int(size * 2 + 1) * 5   # This must be an odd number
+        self.max_length = int(size * 2 + 1) * 3  # This must be an odd number
+        self.has_mapped = {}
+        self.curX = None
+        self.curY = None
+        self.size = size
+        self.location = location or caller.location
+
+    def create_grid(self):
+        # Create an empty grid of the configured size
+        board = []
+        for row in range(self.max_length):
+            board.append([])
+            for column in range(int(self.max_width/5)):
+                board[row].extend([' ', '   ', ' '])
+        return board
+
+    def exit_name_as_ordinal(self, ex):
+        exit_name = ex.name
+        if exit_name not in _COMPASS_DIRECTIONS:
+            compass_aliases = [direction in ex.aliases.all() for direction in _COMPASS_DIRECTIONS.keys()]
+            if compass_aliases[0]:
+                exit_name = compass_aliases[0]
+            if exit_name not in _COMPASS_DIRECTIONS:
+                return ''
+        return exit_name
+
+    def update_pos(self, room, exit_name):
```

Comment:
Example of the needed doc string to add: 
```python
Update the position pointer.

Args: 
    room (Room): The current location. 
    exit_name (str): The name of the exit to to use in this room. This must 
        be a valid compass direction, or an error will be raised. 
Raises: 
    KeyError: If providing a non-compass exit name.

```

This is parsed automatically and becomes part of the documentation API docs, so the style needs to be followed.